java/ingest/src/main/java/sleeper/ingest/IngestRecordsUsingPropertiesSpecifiedMethod.java

```diff
@@ -121,49 +110,46 @@ public static IngestResult ingestFromRecordIterator(
      */
-    public static IngestCoordinator<Record> createIngestCoordinatorWithProperties(
+    private static IngestCoordinator<Record> createIngestCoordinatorWithProperties(
             IngestProperties ingestProperties,
             InstanceProperties instanceProperties,
-            BufferAllocator bufferAllocator,
             S3AsyncClient s3AsyncClient) {
         // Define a factory function for record batches
         String recordBatchType = instanceProperties.get(INGEST_RECORD_BATCH_TYPE).toLowerCase(Locale.ROOT);
         String fileWriterType = instanceProperties.get(INGEST_PARTITION_FILE_WRITER_TYPE).toLowerCase(Locale.ROOT);
-        StandardIngestCoordinator.BackedBuilder ingestCoordinatorBuilder;
+        StandardIngestCoordinator.Builder builder = StandardIngestCoordinator.builder().fromProperties(ingestProperties);
         if (recordBatchType.equals("arraylist")) {
-            ingestCoordinatorBuilder = StandardIngestCoordinator.builder().fromProperties(ingestProperties)
-                    .backedByArrayList()
-                    .maxNoOfRecordsInMemory((int) ingestProperties.getMaxInMemoryBatchSize())
-                    .maxNoOfRecordsInLocalStore(ingestProperties.getMaxRecordsToWriteLocally());
+            builder.arrayListRecordBatchFactory(
+                    (int) ingestProperties.getMaxInMemoryBatchSize(),
+                    ingestProperties.getMaxRecordsToWriteLocally());
         } else if (recordBatchType.equals("arrow")) {
-            ingestCoordinatorBuilder = StandardIngestCoordinator.builder().fromProperties(ingestProperties)
-                    .backedByArrow()
-                    .arrowBufferAllocator(bufferAllocator)
-                    .maxNoOfRecordsToWriteToArrowFileAtOnce(instanceProperties.getInt(ARROW_INGEST_MAX_SINGLE_WRITE_TO_FILE_RECORDS))
-                    .workingArrowBufferAllocatorBytes(instanceProperties.getLong(ARROW_INGEST_WORKING_BUFFER_BYTES))
-                    .minBatchArrowBufferAllocatorBytes(instanceProperties.getLong(ARROW_INGEST_BATCH_BUFFER_BYTES))
-                    .maxBatchArrowBufferAllocatorBytes(instanceProperties.getLong(ARROW_INGEST_BATCH_BUFFER_BYTES))
-                    .maxNoOfBytesToWriteLocally(ingestProperties.getMaxRecordsToWriteLocally());
+            builder.arrowRecordBatchFactory(arrowBuilder -> arrowBuilder
+                    .maxNoOfRecordsToWriteToArrowFileAtOnce(
+                            instanceProperties.getInt(ARROW_INGEST_MAX_SINGLE_WRITE_TO_FILE_RECORDS))
+                    .workingBufferAllocatorBytes(instanceProperties.getLong(ARROW_INGEST_WORKING_BUFFER_BYTES))
+                    .minBatchBufferAllocatorBytes(instanceProperties.getLong(ARROW_INGEST_BATCH_BUFFER_BYTES))
+                    .maxBatchBufferAllocatorBytes(instanceProperties.getLong(ARROW_INGEST_BATCH_BUFFER_BYTES))
+                    .maxNoOfBytesToWriteLocally(ingestProperties.getMaxRecordsToWriteLocally()));
```

Comment:
This is incorrect, the records to write locally and bytes to write locally are two separate things.